torch/_inductor/ir.py

```diff
@@ -4759,20 +4824,15 @@ def find_device(tensor_args, example_output):
         return None
 
     def has_side_effects(self):
```

Comment:
Unrelated to this PR: I want us to kill has_side_effects. We should detect this, not mark it.

cc: @eellison @Chillee